### PR TITLE
Hotfix: Don't send members for unpermitted channels

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Hot fix for [this](https://github.com/status-im/status-desktop/issues/12114#issuecomment-1721426909) problem.

Important changes:
- [x] Empty community channels member list, if there is no permissions settled up for that channels when sending a community description message
- [x] Restore community channels member list from community member list if there is no permissions settled up

Actual fix should be implemented here: https://github.com/status-im/status-desktop/issues/12188

### CODE REVIEW

File: protocol/communities/community.go [hunk 1]
```diff
@@ -1231,7 +1231,16 @@ func (o *Community) Description() *protobuf.CommunityDescription {
 }
 
 func (o *Community) marshaledDescription() ([]byte, error) {
-	return proto.Marshal(o.config.CommunityDescription)
+	// Clear members list for each channel what don't have permissions
```
Comment (jrainville) [inline protocol/communities/community.go]:
```suggestion
	// Clear members list for channels that don't have permissions
```

File: protocol/communities/manager.go [hunk 2]
```diff
@@ -1272,6 +1272,14 @@ func (m *Manager) HandleCommunityDescriptionMessage(signer *ecdsa.PublicKey, des
 }
 
 func (m *Manager) handleCommunityDescriptionMessageCommon(community *Community, description *protobuf.CommunityDescription, payload []byte) (*CommunityResponse, error) {
+	// For each channel what don't have permissions, we use community members list instead
```
Comment (jrainville) [inline protocol/communities/manager.go]:
```suggestion
	// For each channel that doesn't have permissions, we use community members list instead
```

**Review by jrainville — APPROVED** [review]

**Review by osmaczko — APPROVED** [review]

File: protocol/communities/community.go (hunk 1 above)
Comment (MishkaRogachev) [reply protocol/communities/community.go]: fixed

File: protocol/communities/manager.go (hunk 2 above)
Comment (MishkaRogachev) [reply protocol/communities/manager.go]: fixed

**Pull request merged**